pkg/config/config_test.go

```
@@ -583,6 +583,37 @@ func TestGetFIMPathConfigs(t *testing.T) {
+	viper.Reset()
+	dir := t.TempDir()
+	cfg := `{"eventDedup": {"enabled": true, "slotsExponent": 18}, "ruleCooldown": {"ruleCooldownOnProfileFailure": true}, "fim": {"dedupConfig": {"dedupEnabled": true}}}`
+	require.NoError(t, os.WriteFile(dir+"/config.json", []byte(cfg), 0644))
+
+	config, err := LoadConfigOptional(dir, false)
+	require.NoError(t, err)
+
+	assert.False(t, config.AlertDeduplication.Bypass)
+	assert.True(t, config.EventDedup.Enabled)
+	assert.False(t, config.RuleCoolDown.Disabled)
+	assert.True(t, config.RuleCoolDown.OnProfileFailure)
```

golangci-lint (2.12.2)

[error] SA1019: config.RuleCoolDown.OnProfileFailure is deprecated: retained for backward compatibility. Setting it false still disables cooldown, but new callers should use Disabled as the explicit master switch. 

(staticcheck)